src/collections/maple_tree/mod.rs

```diff
@@ -273,5 +742,285 @@ mod tests {
+    // inserting the pivots and values manually
+    #[test]
+    fn test_range_node_move_pivots_right(){
+        
+        let mut rn: RangeLeaf<usize, usize> = RangeLeaf::new();
+        rn.pivot[0] = M::Some(6);
+         
+        assert!(rn.move_pivots_right(0, 2) == true);
+        for i in 0..R_CAPACITY{
+            if i == 2{
+                assert!(rn.pivot[2] == M::Some(6));
+            }
+            else{
+                assert!(rn.pivot[i] == M::None);
+            }
+        }
+
+        let testOne: [usize; R_CAPACITY] = [1, 2, 6, 7, 0, 17, 19];  
+
+        rn.pivot[0] = M::Some(1);
+        rn.pivot[1] = M::Some(2); 
+        rn.pivot[2] = M::Some(6);
+        rn.pivot[3] = M::Some(7);
+        rn.pivot[4] = M::Some(17);
+        rn.pivot[5] = M::Some(19);
+        rn.pivot[6] = M::None;
+        
+        assert!(rn.move_pivots_right(4, 1) == true);
+
+
+        for i in 0..R_CAPACITY{
+            if testOne[i] == 0{
+                assert!(rn.pivot[i] == M::None);
+            }
+            else{
+                println!("{}", i);
+                assert!(rn.pivot[i] == M::Some(testOne[i]));
+            }
+        }
+        
+        let testTwo: [usize; R_CAPACITY] = [1, 2, 6, 7, 10, 15, 20];
+
+
+        for i in 0..R_CAPACITY{
+            rn.pivot[i] = M::Some(testTwo[i]);
+        }
+
+        assert!(rn.move_pivots_right(4,1) == false);
+
+        let testThree: [usize; R_CAPACITY] = [1, 2, 6, 7, 10, 15, 0];
+
+        for i in 0.. R_CAPACITY-1{
+            if testThree[i] == 0{
+                rn.pivot[i] = M::None;
+            }
+            else{
+                rn.pivot[i] = M::Some(testThree[i]);
+            }            
+        }
+
+        assert!(rn.move_pivots_right(2, 3) == false);
+
+    }
+    
+    fn check_node_state(pivots: [usize; R_CAPACITY], values: [usize; CAPACITY], node: &RangeLeaf<usize, usize> ){
+        
+        print!("Pivots - [");
+        for i in 0..R_CAPACITY{
+            if(pivots[i] == 0){
+                print!(", X"); 
+                assert!(M::None == node.pivot[i]);
+            }
+            else{
+                match node.pivot[i]{
+                    M::Some(p) => { print!(", {}", p) },
+                    _ => {print!(", X")}
+                }
+                assert!(M::Some(pivots[i]) == node.pivot[i]); 
+            }
+        }
+        
+        print!("]\n Values - [");
+        for i in 0..CAPACITY{
+            if(values[i] == 0){
+                print!(", X");
+                assert!(M::None == node.value[i]);
+            }
+            else{
+                match node.value[i]{
+                    M::Some(p) => { print!(", {}", p) },
+                    _ => {print!(", X")}
+                }
+             
+                assert!(M::Some(values[i]) == node.value[i]);
+            }
+        }
+        print!("]\n\n");
+    }
+
+    fn print_range_node_state(rn: &RangeLeaf<usize, usize>){
```

Comment:
You should look at https://doc.rust-lang.org/std/fmt/trait.Display.html, if you implement that trait, then the following works:

println!("{}", node);

And so you'll gain the ability to print the node anytime, without a dedicated function.

Reply:
This is done! I completely forgot you could do this haha